src/hivpy/prep.py

```diff
@@ -91,6 +111,97 @@ def get_suspect_risk_pop(self, pop: Population):
                                    COND(col.LTP_HIV_STATUS, op.eq, True),
                                    COND(col.R_PREP, op.lt, self.prob_suspect_risk_prep)))
 
+    def get_presumed_hiv_neg_pop(self, pop: Population):
+        """
+        Return the sub-population that has been tested and is HIV positive but
+        received a false negative result.
+        """
+        # FIXME: should these individuals also be fixed instead of rerolling false negatives?
+        false_neg_pop = pop.get_sub_pop(AND(COND(col.EVER_TESTED, op.eq, True),
+                                            COND(col.HIV_STATUS, op.eq, True)))
+
+        # general test sensitivity
+        eff_test_sens = pop.hiv_diagnosis.test_sens_general
+        if not pop.hiv_diagnosis.init_prep_inj_na:
+            # FIXME: does this have to be exactly 3 months ago or can it be <=?
+            recently_infected_pop = pop.get_sub_pop_intersection(
+                pop.get_sub_pop(COND(col.DATE_HIV_INFECTION, op.eq, pop.date - date(0.25))), false_neg_pop)
+
+            # expand sensitivity into a list
+            eff_test_sens = [pop.hiv_diagnosis.test_sens_general] * len(false_neg_pop)
+            false_neg_list = list(false_neg_pop)
+            # find indices in false_neg_pop that correspond to people belonging to recently_infected_pop
+            common_i = [false_neg_list.index(i) for i in false_neg_list if i in recently_infected_pop]
+
+            # FIXME: is there a better way to do this?
+            for i in common_i:
+                # lower test sensitivity used to mimic more people starting prep when they have hiv
+                eff_test_sens[i] = pop.hiv_diagnosis.test_sens_primary_ab
+
+        # false negative outcomes
+        r = rng.uniform(size=len(false_neg_pop))
+        mask = r > eff_test_sens
+
+        return pop.apply_bool_mask(mask, false_neg_pop)
+
+    def prep_willingness(self, pop: Population):
```

Comment:
We need to mimic when the preferences are set in the SAS code; for each kind of prep the preference is set _either_ when the type of prep is introduced (if already over 15) or when the person turns 15 (if they turn 15 after the prep is introduced)
`if (caldate{t} = date_prep_oral_intro > . and age ge 15) or (age = 15 and caldate{t} >= date_prep_oral_intro > .) then do;`

This can make a difference because the preference distributions can shift over time due to e.g. option code

Reply:
+ add to evolve near the beginning of a time step

Reply:
Fixed the above issues + moved PrEP usage code - PR should be mergeable now!